### DISCUSSION TITLE
Add new segmentation label into pre-trained model (wholeBrainSeg Large UNEST segmentation).

### DISCUSSION THREAD

**Discussion by anuragp2018** [Project-MONAI/model-zoo#590] [General]
Hello,

I want to add a new segmentation label to the WholeBrainSeg Large UNET segmentation model and retrain the model. Can you please help or guide me on how to do that?

**Comment by ericspod** [comment]
Hi @anuragp2018, I would suggest starting with the [segmentation template bundle](https://github.com/Project-MONAI/model-zoo/tree/dev/models/segmentation_template). If your model is just a UNet network then the template would require minimal modification to become your bundle. Copy the whole bundle directory, modify the configurations and documentation in README.md, provide a license, and whatever other code you need in the scripts directory. From there you can propose a PR and we'll help with anything more that needs doing.

**Reply by anuragp2018** [reply]
Actually, I don't have any model. I just want to add a new segmentation label to a pre-trained model and retrain the model.

**Reply by ericspod** [reply]
In this case you're going to want to do some form of transfer learning. One way is to first load the model and then replace [the last block](https://github.com/Project-MONAI/model-zoo/blob/5eb4be59d3d405879dd53aad11e82c1528294eea/models/wholeBrainSeg_Large_UNEST_segmentation/scripts/networks/unest_base_patch_4.py#L197) with one having a different number of output channels, then either fine tune just that block or the whole network on your dataset. [This tutorial](https://github.com/Project-MONAI/tutorials/tree/ecf8eadbc60277aa8bb604f018e1adf8b75b7dbd/model_zoo/transfer_learning_with_bundle) may help with that, otherwise @tangy5 or @yiheng-wang-nv may have some insights as well.
